cpu/stm32/periph/eth.c

```diff
@@ -166,9 +178,71 @@ static void _init_buffer(void)
     ETH->DMATDLAR = (uintptr_t)&tx_desc[0];
 }
 
+static int stm32_eth_set(netdev_t *dev, netopt_t opt,
+                         const void *value, size_t max_len)
+{
+    int res = -1;
+
+    switch (opt) {
+    case NETOPT_ADDRESS:
+        assert(max_len >= ETHERNET_ADDR_LEN);
+        stm32_eth_set_mac((char *)value);
+        res = ETHERNET_ADDR_LEN;
+        break;
+    default:
+        res = netdev_eth_set(dev, opt, value, max_len);
+        break;
+    }
+
+    return res;
+}
+
+static int stm32_eth_get(netdev_t *dev, netopt_t opt,
+                         void *value, size_t max_len)
+{
+    int res = -1;
+
+    switch (opt) {
+    case NETOPT_ADDRESS:
+        assert(max_len >= ETHERNET_ADDR_LEN);
+        stm32_eth_get_mac((char *)value);
+        res = ETHERNET_ADDR_LEN;
+        break;
+    case NETOPT_LINK:
+        res = (_phy_read(0, PHY_BSMR) & BSMR_LINK_STATUS);
+        break;
+    default:
+        res = netdev_eth_get(dev, opt, value, max_len);
+        break;
+    }
+
+    return res;
+}
+
+#if IS_USED(MODULE_STM32_ETH_LINK_UP)
+static void _timer_cb(void *arg)
+{
+    netdev_t *dev = (netdev_t *)arg;
+    if (stm32_eth_get(dev, NETOPT_LINK, NULL, 0)) {
+        _link_status = 1;
+        dev->event_callback(dev, NETDEV_EVENT_ISR);
```

Comment:
If this causes `stm32_eth_isr` to be called, it will also generate a `NETDEV_EVENT_RX_COMPLETE` event.

Reply:
I’m not quite sure how to handle this right now. In the old `eth.c` implementation there was this function` stm32_eth_get_rx_status_owned`, which was used to check whether a frame reception was completed or not. 
What is the best way to check for that in the current implementation?